internal/grpc/services/ocmshareprovider/ocmshareprovider.go

```diff
@@ -210,13 +214,38 @@ func (s *service) getWebappProtocol(share *ocm.Share) *ocmd.Webapp {
 }
 
 func (s *service) getDataTransferProtocol(ctx context.Context, share *ocm.Share) *ocmd.Datatx {
-	// TODO discover the size
+	var size uint64
+	// get the path of the share
+	statRes, err := s.gateway.Stat(ctx, &providerpb.StatRequest{
+		Ref: &providerpb.Reference{
+			ResourceId: share.ResourceId,
+		},
+	})
+	if err != nil {
+		panic(err)
```

Comment:
Shouldn't we just fail the request here rather than killing the server?

Reply:
`Panic` doesn't seem to kill the server but the current execute flow indeed.
In this case I was kind of following `getWebappProtocol`, and it sure is bad if this particular request would fail. And it also meant I didn't need to add error handling further up.
But (maybe like you ... ?) I'm actually not too fond of this `panic` reaction which is used at more places. Is there a protocol for when we should use this?

Reply:
I appreciate we agree on the principle ;-) let me check what we do elsewhere, I don't think there's a strict "protocol" for using `panic()` vs logging the error - I'd go for the latter indeed (and we know that logging could be vastly improved in several corners of the code base).

Reply:
If at that point of the code you expect to find the transfer protocol, for example because you have already checked before, is fine